File: VB/MultiPaneExtension/MultiPaneSettings.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Text.Json
Imports System.Threading.Tasks

Namespace MultiPaneExtension
    Friend Class MultiPaneSettings
        Public Property MultiPaneEnabled() As Boolean
        Public Property ShowPaneTitles() As Boolean
        Public Property AllowPaneCollapsing() As Boolean
        Public Property UseGridLayout() As Boolean

        Public Sub New()
            MultiPaneEnabled = False
            ShowPaneTitles = True
            AllowPaneCollapsing = True
            UseGridLayout = True
        End Sub

        Public Shared Function FromJson(ByVal json As String) As MultiPaneSettings
            If String.IsNullOrEmpty(json) Then
                Return New MultiPaneSettings()
            End If
            Return TryCast(JsonSerializer.Deserialize(Of MultiPaneSettings)(json), MultiPaneSettings)
        End Function

        Public Function ToJson() As String
            Return JsonSerializer.Serialize(Me)
        End Function


    End Class
End Namespace

File: CS/MultiPaneExtension/MultiPaneSettings.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MultiPaneExtension
{
    class MultiPaneSettings
    {
        public bool MultiPaneEnabled { get; set; }
        public bool ShowPaneTitles { get; set; }
        public bool AllowPaneCollapsing { get; set; }
        public bool UseGridLayout { get; set; }

        public MultiPaneSettings()
        {
            MultiPaneEnabled = false;
            ShowPaneTitles = true;
            AllowPaneCollapsing = true;
            UseGridLayout = true;
        }

        public static MultiPaneSettings FromJson(string json)
        {
            if (string.IsNullOrEmpty(json))
                return new MultiPaneSettings();
            return JsonConvert.DeserializeObject<MultiPaneSettings>(json) as MultiPaneSettings;
        }

        public string ToJson()
        {
            return JsonConvert.SerializeObject(this);
        }


    }
}
